crates/ruff_linter/src/rules/airflow/rules/moved_to_provider_in_3.rs

```diff
@@ -937,22 +937,47 @@ fn check_names_moved_to_provider(checker: &Checker, expr: &Expr, ranged: TextRan
         ranged.range(),
     );
 
-    if let ProviderReplacement::AutoImport {
-        module,
-        name,
-        provider: _,
-        version: _,
-    } = replacement
-    {
-        diagnostic.try_set_fix(|| {
-            let (import_edit, binding) = checker.importer().get_or_import_symbol(
-                &ImportRequest::import_from(module, name),
-                expr.start(),
-                checker.semantic(),
-            )?;
-            let replacement_edit = Edit::range_replacement(binding, ranged.range());
-            Ok(Fix::safe_edits(import_edit, [replacement_edit]))
-        });
+    let semantic = checker.semantic();
+    match replacement {
+        ProviderReplacement::AutoImport {
+            module,
+            name,
+            provider: _,
+            version: _,
+        } => {
+            if is_guarded_by_try_except(expr, module, name, semantic) {
+                return;
+            }
+            diagnostic.try_set_fix(|| {
+                let (import_edit, binding) = checker.importer().get_or_import_symbol(
+                    &ImportRequest::import_from(module, name),
+                    expr.start(),
+                    checker.semantic(),
+                )?;
+                let replacement_edit = Edit::range_replacement(binding, ranged.range());
+                Ok(Fix::safe_edits(import_edit, [replacement_edit]))
+            });
+        }
+        ProviderReplacement::SourceModuleMovedToProvider {
+            module,
+            name,
+            provider: _,
+            version: _,
+        } => {
+            if is_guarded_by_try_except(expr, module, name.as_str(), semantic) {
+                return;
+            }
+            diagnostic.try_set_fix(|| {
+                let (import_edit, binding) = checker.importer().get_or_import_symbol(
+                    &ImportRequest::import_from(module, name.as_str()),
+                    expr.start(),
+                    checker.semantic(),
+                )?;
+                let replacement_edit = Edit::range_replacement(binding, ranged.range());
+                Ok(Fix::safe_edits(import_edit, [replacement_edit]))
+            });
+        }
+        _ => {}
     }
```

Comment:
It's a bit unfortunate to duplicate these blocks so many times, but I don't see a super easy way to factor out a helper method. One idea is to extract `module` and `name` first to avoid duplicating the `try_set_fix` calls at least:

```suggestion
    let semantic = checker.semantic();
    if let Some((module, name)) = match &replacement {
        ProviderReplacement::AutoImport { module, name, .. } => Some((module, *name)),
        ProviderReplacement::SourceModuleMovedToProvider { module, name, .. } => {
            Some((module, name.as_str()))
        }
        _ => None,
    } {
        if is_guarded_by_try_except(expr, module, name, semantic) {
            return;
        }
        diagnostic.try_set_fix(|| {
            let (import_edit, binding) = checker.importer().get_or_import_symbol(
                &ImportRequest::import_from(module, name),
                expr.start(),
                checker.semantic(),
            )?;
            let replacement_edit = Edit::range_replacement(binding, ranged.range());
            Ok(Fix::safe_edits(import_edit, [replacement_edit]))
        });
    }
```

Reply:
Yep, I tried a few ways but failed. Thanks for the suggestion. I just updated it to all the rules!